include/riak_kv_ts.hrl

```diff
@@ -83,6 +83,27 @@
           %% since v2
           group_by = ?GROUP_BY_DEFAULT :: [{identifier, binary()}] | [{FieldPos::integer(), FieldName::binary()}]
         }).
+-record(riak_select_v3,
+        {
+          'SELECT'              :: #riak_sel_clause_v1{},
+          'FROM'        = <<>>  :: binary() | {list, [binary()]} | {regex, list()},
+          'WHERE'       = []    :: [riak_ql_ddl:filter()],
+          'ORDER BY'            :: undefined | [riak_kv_qry_compiler:sorter()],
+          'LIMIT'               :: undefined | riak_kv_qry_compiler:limit(),
+          'OFFSET'              :: undefined | riak_kv_qry_compiler:offset(),
+          helper_mod            :: atom(),
+          %% will include groups when we get that far
+          partition_key = none  :: none | #key_v1{},
+          %% indicates whether this query has already been compiled to a sub query
+          is_executable = false :: boolean(),
+          type          = sql   :: sql | timeseries,
+          cover_context = undefined :: term(), %% for parallel queries
+          allow_qbuf_reuse = false  :: boolean(),  %% control reuse of query buffers
+          %% prolly a mistake to put this here - should be in DDL
+          local_key,
+          %% since v2
+          group_by = ?GROUP_BY_DEFAULT :: [{identifier, binary()}] | [{FieldPos::integer(), FieldName::binary()}]
```

Comment:
I agree in general but I would rather leave the rename to @andytill. It will obviously require a sweeping change all over the place. The current PR's diff is already big enough.